tests/unit/nn/test_transforms.py

```diff
@@ -0,0 +1,157 @@
+    return GraphTransform(V_transform=scale_transform, E_transform=scale_transform)
+
+
+@pytest.fixture
+def X():
+    return torch.tensor([[99.0, 99.0, 1.0, 2.0], [99.0, 99.0, 3.0, 4.0]])
+
+
+@pytest.fixture
+def prediction():
+    return torch.tensor([[1.0, 2.0]])
+
+
+@pytest.fixture
+def variance():
+    return torch.tensor([[0.1, 0.2]])
+
+
+@pytest.fixture
+def bmg():
+    V = torch.tensor([[99.0, 99.0, 1.0, 2.0], [99.0, 99.0, 3.0, 4.0]])
+    E = torch.tensor([[99.0, 99.0, 1.0, 2.0], [99.0, 99.0, 3.0, 4.0]])
+    return MockBatchMolGraph(V=V, E=E)
+
+
+def test_uneven_shapes():
+    with pytest.raises(ValueError):
+        ScaleTransform(mean=[0.0], scale=[1.0, 2.0])
+
+
+def test_padding(mean, scale, pad):
+    scale_transform = ScaleTransform(mean, scale, pad)
+    assert torch.all(scale_transform.mean[0, :pad] == 0.0).item()
+    assert torch.all(scale_transform.scale[0, :pad] == 1.0).item()
+
+
+def test_from_standard_scaler(mean, scale, scaler):
+    scale_transform = ScaleTransform.from_standard_scaler(scaler)
+
+    assert torch.all(scale_transform.mean == torch.tensor([0.0, 1.0])).item()
+    assert torch.all(scale_transform.scale == torch.tensor([2.0, 3.0])).item()
+
+
+def test_scale_transform_forward_train(scale_transform, X):
+    scale_transform.train()
+    output_X = scale_transform(X)
+    assert output_X is X
+
+
+def test_scale_transform_forward_eval(tensor_mean, tensor_scale, scale_transform, X):
+    scale_transform.eval()
+    output_X = scale_transform(X)
+    expected_X = (X - tensor_mean) / tensor_scale
+    assert torch.equal(output_X, expected_X)
+
+
+def test_unscale_transform_forward_train(unscale_transform, X):
+    unscale_transform.train()
+    output_X = unscale_transform(X)
+    assert output_X is X
+
+
+def test_unscale_transform_forward_eval(mean, scale, unscale_transform, prediction):
+    unscale_transform.eval()
+    output = unscale_transform(prediction)
+    expected = prediction * scale + mean
+    print(output, expected)
+    assert torch.equal(output, expected)
+
+
+def test_unscale_transform_variance_train(unscale_transform, variance):
+    unscale_transform.train()
+    output_variance = unscale_transform.transform_variance(variance)
+    assert output_variance is variance
+
+
+def test_unscale_transform_variance_eval(scale, unscale_transform, variance):
+    unscale_transform.eval()
+    output_variance = unscale_transform.transform_variance(variance)
+    expected_variance = variance * scale**2
+    assert torch.allclose(output_variance, expected_variance.float())
+
+
+def test_graph_transform_forward_train(graph_transform, bmg):
+    graph_transform.train()
+    output_bmg = graph_transform(bmg)
+    assert output_bmg is bmg
+
+
+def test_graph_transform_forward_eval(graph_transform, bmg):
+    graph_transform.eval()
+    expected_V = graph_transform.V_transform(bmg.V)
+    expected_E = graph_transform.E_transform(bmg.E)
+
+    transformed_bmg = graph_transform(bmg)
```

Comment:
`graph_transform` returns a reference to the original array which is changed in place, so `transformed_bmg` is the same object as `bmg`. That is why I use `bmg.V` and `bmg.E` to make the expected outputs before doing the graph transform.

Reply:
How is the array changed in place? The operations should be out-of-place, no?

Reply:
The forward of GraphTransform doesn't make a new BatchMolGraph. It changes the `V` and `E` attributes of the input `bmg` and returns that object. 
```
def forward(self, bmg: BatchMolGraph) -> BatchMolGraph:
        if self.training:
            return bmg

        bmg.V = self.V_transform(bmg.V)
        bmg.E = self.E_transform(bmg.E)

        return bmg
```

Reply:
I see- I think I wasn't clear on your wording. I think you mean to say that it modifies the `BatchMolGraph` in-place rather than it modifies the array in-place. Modifying the array in-place would mean changing the tensor values at the exact same address in memory, which we don't do because we use out-of-place operations in the underlying transforms.
```python
X: Tensor = ...
Y: Tensor = ...
# out of place
X + Y
# which is syntactic sugar for this
X.add(Y)
# but this is in-place
X += Y
# which is syntactic sugar for this
X.add_(Y)
```

Reply:
Yes sorry about that. When I said "reference to the original array" I really meant "reference to the original bmg object".